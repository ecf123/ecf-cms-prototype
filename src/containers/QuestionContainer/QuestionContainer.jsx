import AddQuestion from "../../components/AddQuestion/AddQuestion";
import "./QuestionContainer.scss";

const QuestionContainer = ({ questionsArray, handleDelete }) => {
  const questionsArrayJSX = questionsArray?.map((question, index) => {
    return (
      <div key={index}>
        <AddQuestion
          question={question.question}
          answers={question.answers}
          questionNumber={question.questionNumber}
          handleDelete={handleDelete}
        />
        {index < questionsArray.length - 1 && (
          <div className="container__line"></div>
        )}
      </div>
    );
  });

  return <div className="container">{questionsArrayJSX}</div>;
};

export default QuestionContainer;
