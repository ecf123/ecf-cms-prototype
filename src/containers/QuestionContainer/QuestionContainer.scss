@use "../../styles/variables.scss" as *;

.container {
  width: 27.625rem;
  height: 41.75rem;
  display: flex;
  flex-direction: column;
  background: $color-white-smoke;
  overflow-y: scroll;
  padding: 2.18rem;

  // &__question {

  // }

  &__line {
    margin-top: 2.18rem;
    margin-bottom: 2.18rem;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;
    border: 0.5px solid $color-grey-light;
    width: 100%;
    background-color: $color-grey-light;
  }
}
