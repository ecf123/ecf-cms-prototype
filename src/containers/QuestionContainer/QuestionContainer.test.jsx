import { customRender } from "../../utils/testUtils";
import QuestionContainer from "./QuestionContainer";

const questionsArray = [
  {
    question: "What sells on Amazon the most",
    answers: [
      "Grenade Protein Bars *",
      "pHD Smart Plan Protein Bar *",
      "Professional Diva HairDryer",
      "Black Plus Decker Heater",
      "Serious Mass Protein Powder *",
    ],
    questionNumber: 1,
  },
  {
    question: "What sells on Amazon the most",
    answers: [
      "Grenade Protein Bars *",
      "pHD Smart Plan Protein Bar *",
      "Professional Diva HairDryer",
      "Black Plus Decker Heater",
      "Serious Mass Protein Powder *",
    ],
    questionNumber: 2,
  },
  {
    question: "What sells on Amazon the most",
    answers: [
      "Grenade Protein Bars *",
      "pHD Smart Plan Protein Bar *",
      "Professional Diva HairDryer",
      "Black Plus Decker Heater",
      "Serious Mass Protein Powder *",
    ],
    questionNumber: 3,
  },
  {
    question: "What sells on Amazon the most",
    answers: [
      "Grenade Protein Bars *",
      "pHD Smart Plan Protein Bar *",
      "Professional Diva HairDryer",
      "Black Plus Decker Heater",
      "Serious Mass Protein Powder *",
    ],
    questionNumber: 4,
  },
  {
    question: "What sells on Amazon the most",
    answers: [
      "Grenade Protein Bars *",
      "pHD Smart Plan Protein Bar *",
      "Professional Diva HairDryer",
      "Black Plus Decker Heater",
      "Serious Mass Protein Powder *",
    ],
    questionNumber: 5,
  },
  {
    question: "What sells on Amazon the most",
    answers: [
      "Grenade Protein Bars *",
      "pHD Smart Plan Protein Bar *",
      "Professional Diva HairDryer",
      "Black Plus Decker Heater",
      "Serious Mass Protein Powder *",
    ],
    questionNumber: 6,
  },
];

it("Should match the snapshot, with the data inserted", () => {
  const { container } = customRender(
    <QuestionContainer questionsArray={questionsArray} />
  );
  expect(container).toMatchSnapshot();
});
