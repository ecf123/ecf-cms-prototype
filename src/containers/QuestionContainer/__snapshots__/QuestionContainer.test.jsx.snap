// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match the snapshot, with the data inserted 1`] = `
<div>
  <div
    class="container"
  >
    <div>
      <div
        class="question"
      >
        <div
          class="question__container"
        >
          <h6
            class="question__title"
          >
            Q
            1
            . 
            What sells on Amazon the most
            ?
             
          </h6>
          <ol
            class="question__answers"
            type="A"
          >
            <li
              class="question__each-answer"
            >
              Grenade Protein Bars 
               
              <b
                class="question__correct"
              >
                (correct)
              </b>
            </li>
            <li
              class="question__each-answer"
            >
              pHD Smart Plan Protein Bar 
               
              <b
                class="question__correct"
              >
                (correct)
              </b>
            </li>
            <li
              class="question__each-answer"
            >
              Professional Diva HairDryer
            </li>
            <li
              class="question__each-answer"
            >
              Black Plus Decker Heater
            </li>
            <li
              class="question__each-answer"
            >
              Serious Mass Protein Powder 
               
              <b
                class="question__correct"
              >
                (correct)
              </b>
            </li>
          </ol>
        </div>
        <div
          class="buttons"
        >
          <img
            alt="Delete Question"
            class="buttons__bin"
            src="bin-icon.svg"
          />
        </div>
      </div>
      <div
        class="container__line"
      />
    </div>
    <div>
      <div
        class="question"
      >
        <div
          class="question__container"
        >
          <h6
            class="question__title"
          >
            Q
            2
            . 
            What sells on Amazon the most
            ?
             
          </h6>
          <ol
            class="question__answers"
            type="A"
          >
            <li
              class="question__each-answer"
            >
              Grenade Protein Bars 
               
              <b
                class="question__correct"
              >
                (correct)
              </b>
            </li>
            <li
              class="question__each-answer"
            >
              pHD Smart Plan Protein Bar 
               
              <b
                class="question__correct"
              >
                (correct)
              </b>
            </li>
            <li
              class="question__each-answer"
            >
              Professional Diva HairDryer
            </li>
            <li
              class="question__each-answer"
            >
              Black Plus Decker Heater
            </li>
            <li
              class="question__each-answer"
            >
              Serious Mass Protein Powder 
               
              <b
                class="question__correct"
              >
                (correct)
              </b>
            </li>
          </ol>
        </div>
        <div
          class="buttons"
        >
          <img
            alt="Delete Question"
            class="buttons__bin"
            src="bin-icon.svg"
          />
        </div>
      </div>
      <div
        class="container__line"
      />
    </div>
    <div>
      <div
        class="question"
      >
        <div
          class="question__container"
        >
          <h6
            class="question__title"
          >
            Q
            3
            . 
            What sells on Amazon the most
            ?
             
          </h6>
          <ol
            class="question__answers"
            type="A"
          >
            <li
              class="question__each-answer"
            >
              Grenade Protein Bars 
               
              <b
                class="question__correct"
              >
                (correct)
              </b>
            </li>
            <li
              class="question__each-answer"
            >
              pHD Smart Plan Protein Bar 
               
              <b
                class="question__correct"
              >
                (correct)
              </b>
            </li>
            <li
              class="question__each-answer"
            >
              Professional Diva HairDryer
            </li>
            <li
              class="question__each-answer"
            >
              Black Plus Decker Heater
            </li>
            <li
              class="question__each-answer"
            >
              Serious Mass Protein Powder 
               
              <b
                class="question__correct"
              >
                (correct)
              </b>
            </li>
          </ol>
        </div>
        <div
          class="buttons"
        >
          <img
            alt="Delete Question"
            class="buttons__bin"
            src="bin-icon.svg"
          />
        </div>
      </div>
      <div
        class="container__line"
      />
    </div>
    <div>
      <div
        class="question"
      >
        <div
          class="question__container"
        >
          <h6
            class="question__title"
          >
            Q
            4
            . 
            What sells on Amazon the most
            ?
             
          </h6>
          <ol
            class="question__answers"
            type="A"
          >
            <li
              class="question__each-answer"
            >
              Grenade Protein Bars 
               
              <b
                class="question__correct"
              >
                (correct)
              </b>
            </li>
            <li
              class="question__each-answer"
            >
              pHD Smart Plan Protein Bar 
               
              <b
                class="question__correct"
              >
                (correct)
              </b>
            </li>
            <li
              class="question__each-answer"
            >
              Professional Diva HairDryer
            </li>
            <li
              class="question__each-answer"
            >
              Black Plus Decker Heater
            </li>
            <li
              class="question__each-answer"
            >
              Serious Mass Protein Powder 
               
              <b
                class="question__correct"
              >
                (correct)
              </b>
            </li>
          </ol>
        </div>
        <div
          class="buttons"
        >
          <img
            alt="Delete Question"
            class="buttons__bin"
            src="bin-icon.svg"
          />
        </div>
      </div>
      <div
        class="container__line"
      />
    </div>
    <div>
      <div
        class="question"
      >
        <div
          class="question__container"
        >
          <h6
            class="question__title"
          >
            Q
            5
            . 
            What sells on Amazon the most
            ?
             
          </h6>
          <ol
            class="question__answers"
            type="A"
          >
            <li
              class="question__each-answer"
            >
              Grenade Protein Bars 
               
              <b
                class="question__correct"
              >
                (correct)
              </b>
            </li>
            <li
              class="question__each-answer"
            >
              pHD Smart Plan Protein Bar 
               
              <b
                class="question__correct"
              >
                (correct)
              </b>
            </li>
            <li
              class="question__each-answer"
            >
              Professional Diva HairDryer
            </li>
            <li
              class="question__each-answer"
            >
              Black Plus Decker Heater
            </li>
            <li
              class="question__each-answer"
            >
              Serious Mass Protein Powder 
               
              <b
                class="question__correct"
              >
                (correct)
              </b>
            </li>
          </ol>
        </div>
        <div
          class="buttons"
        >
          <img
            alt="Delete Question"
            class="buttons__bin"
            src="bin-icon.svg"
          />
        </div>
      </div>
      <div
        class="container__line"
      />
    </div>
    <div>
      <div
        class="question"
      >
        <div
          class="question__container"
        >
          <h6
            class="question__title"
          >
            Q
            6
            . 
            What sells on Amazon the most
            ?
             
          </h6>
          <ol
            class="question__answers"
            type="A"
          >
            <li
              class="question__each-answer"
            >
              Grenade Protein Bars 
               
              <b
                class="question__correct"
              >
                (correct)
              </b>
            </li>
            <li
              class="question__each-answer"
            >
              pHD Smart Plan Protein Bar 
               
              <b
                class="question__correct"
              >
                (correct)
              </b>
            </li>
            <li
              class="question__each-answer"
            >
              Professional Diva HairDryer
            </li>
            <li
              class="question__each-answer"
            >
              Black Plus Decker Heater
            </li>
            <li
              class="question__each-answer"
            >
              Serious Mass Protein Powder 
               
              <b
                class="question__correct"
              >
                (correct)
              </b>
            </li>
          </ol>
        </div>
        <div
          class="buttons"
        >
          <img
            alt="Delete Question"
            class="buttons__bin"
            src="bin-icon.svg"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
